src/sentry/static/sentry/app/components/avatarCropper.tsx

```diff
@@ -0,0 +1,464 @@
+
+    const diff = this['getDiff' + resizeDirection!.toUpperCase()](yDiff, xDiff);
+
+    let height = container.clientHeight - oldDimensions.top;
+    let width = container.clientWidth - oldDimensions.left;
+
+    // Depending on the direction, we update different dimensions:
+    // nw: size, top, left
+    // ne: size, top
+    // sw: size, left
+    // se: size
+    const editingTop = resizeDirection === 'nw' || resizeDirection === 'ne';
+    const editingLeft = resizeDirection === 'nw' || resizeDirection === 'sw';
+
+    const newDimensions = {
+      top: 0,
+      left: 0,
+      size: oldDimensions.size + diff,
+    };
+
+    if (editingTop) {
+      newDimensions.top = oldDimensions.top - diff;
+      height = container.clientHeight - newDimensions.top;
+    }
+
+    if (editingLeft) {
+      newDimensions.left = oldDimensions.left - diff;
+      width = container.clientWidth - newDimensions.left;
+    }
+
+    if (newDimensions.top < 0) {
+      newDimensions.size = newDimensions.size + newDimensions.top;
+      if (editingLeft) {
+        newDimensions.left = newDimensions.left - newDimensions.top;
+      }
+      newDimensions.top = 0;
+    }
+
+    if (newDimensions.left < 0) {
+      newDimensions.size = newDimensions.size + newDimensions.left;
+      if (editingTop) {
+        newDimensions.top = newDimensions.top - newDimensions.left;
+      }
+      newDimensions.left = 0;
+    }
+
+    const maxSize = Math.min(width, height);
+    if (newDimensions.size > maxSize) {
+      if (editingTop) {
+        newDimensions.top = newDimensions.top + newDimensions.size - maxSize;
+      }
+      if (editingLeft) {
+        newDimensions.left = newDimensions.left + newDimensions.size - maxSize;
+      }
+      newDimensions.size = maxSize;
+    } else if (newDimensions.size < this.MIN_DIMENSION) {
+      if (editingTop) {
+        newDimensions.top = newDimensions.top + newDimensions.size - this.MIN_DIMENSION;
+      }
+      if (editingLeft) {
+        newDimensions.left = newDimensions.left + newDimensions.size - this.MIN_DIMENSION;
+      }
+      newDimensions.size = this.MIN_DIMENSION;
+    }
+
+    return {...oldDimensions, ...newDimensions};
+  };
+
+  validateImage() {
+    const img = this.image.current;
+
+    if (!img) {
+      return null;
+    }
+
+    if (img.naturalWidth < this.MIN_DIMENSION || img.naturalHeight < this.MIN_DIMENSION) {
+      return tct('Please upload an image larger than [size]px by [size]px.', {
+        size: this.MIN_DIMENSION - 1,
+      });
+    }
+
+    if (img.naturalWidth > this.MAX_DIMENSION || img.naturalHeight > this.MAX_DIMENSION) {
+      return tct('Please upload an image smaller than [size]px by [size]px.', {
+        size: this.MAX_DIMENSION,
+      });
+    }
+
+    return null;
+  }
+
+  drawToCanvas() {
+    const canvas = this.canvas.current;
+    if (!canvas) {
+      return;
+    }
+
+    const image = this.image.current;
+    if (!image) {
```

Comment:
@EvanPurkhiser Nit: could we combine these two `if` statements?

Reply:
I somewhat like keeping them separate 🤷‍♂

Reply:
@EvanPurkhiser ok, your call :)